From bbb19069dd10703c4aa06594752d6772fbc6d15d Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:23 -0400
Subject: [ethernet] i40e: use generic vsi_open to unquiesce vsi

Message-id: <1409658042-22689-83-git-send-email-sassmann@redhat.com>
Patchwork-id: 90276
O-Subject: [RHEL7.1 PATCH 082/281] i40e: use generic vsi_open to unquiesce vsi
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shannon Nelson <shannon.nelson@intel.com>

Use the new i40e_vsi_open() for waking VSIs back up in order to
be sure all the standard actions happen.

Change-ID: Ic3479410dd3079733f4951dcea69f101e69e77df
Signed-off-by: Shannon Nelson <shannon.nelson@intel.com>
Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 8276f75748a60a80ab465ad806c8685e624dfb65)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 889ae78..f9f1618 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -3560,7 +3560,7 @@ static void i40e_unquiesce_vsi(struct i40e_vsi *vsi)
  if (vsi->netdev && netif_running(vsi->netdev))
   vsi->netdev->netdev_ops->ndo_open(vsi->netdev);
  else
-  i40e_up(vsi);   /* this clears the DOWN bit */
+  i40e_vsi_open(vsi);   /* this clears the DOWN bit */
 }
 
 /**
-- 
1.7.1